lib/cequel/metal/data_set.rb

```diff
@@ -655,12 +673,16 @@ def row_specifications_cql
 
       protected
 
-      attr_writer :row_limit, :query_consistency
+      attr_writer :row_limit, :query_consistency, :query_page_size, :query_paging_state
 
       private
 
       def execute_cql(cql, *bind_vars)
-        keyspace.execute_with_consistency(cql, bind_vars, query_consistency)
+        keyspace.execute_with_options(cql, bind_vars, {
+          consistency: query_consistency, 
+          page_size: query_page_size, 
+          paging_state: query_paging_state
+        })
```

Comment:
It could, and I think it would clean up (or make unnecessary) the `next_paging_state` if it did, but there is also the question of how the library user would know if the paging_state they're getting is the one they supplied vs the new one.  To some extent, I think that's why these are all prefixed with 'query', they're parameters of the query, not the result.

This is my first foray into the code, so there is a lot of speculation.  I can certainly refactor to have it update if that is the preferred style.